drivers/ethernet/eth_stm32_hal.c

```diff
@@ -1041,6 +968,17 @@ static int eth_initialize(const struct device *dev)
 #if !defined(CONFIG_ETH_STM32_HAL_API_V2)
 	HAL_StatusTypeDef hal_ret = HAL_OK;
 
+	if (!IS_ENABLED(ETH_STM32_AUTO_NEGOTIATION_ENABLE)) {
+		struct phy_link_state state;
+
+		phy_get_link_state(eth_stm32_phy_dev, &state);
+
+		heth->Init.DuplexMode = PHY_LINK_IS_FULL_DUPLEX(state.speed) ? ETH_MODE_FULLDUPLEX
+									     : ETH_HALFDUPLEX_MODE;
```

Comment:
ETH_HALFDUPLEX_MODE is for the API V2
```suggestion
									     : ETH_MODE_HALFDUPLEX;
```